include/bout/region.hxx

```diff
@@ -485,6 +512,37 @@ public:
     return indices.size();
   }
   
+  /// Returns a RegionStats struct desribing the region
+  RegionStats getStats() const {
+    RegionStats result;
+
+    result.numBlocks = blocks.size();
+    if ( result.numBlocks == 0 ) return result;
+    
+    std::vector<int> blockSizes(result.numBlocks);
+    
+    // Get the size of each block using lambda to calculate size
+    std::transform(std::begin(blocks), std::end(blocks), std::begin(blockSizes),
+		   [](const ContiguousBlock &a) { return a.second.ind - a.first.ind;});
+
+    auto minMaxSize = std::minmax_element(std::begin(blockSizes), std::end(blockSizes));
+
+    result.minBlockSize = *(minMaxSize.first); //Note have to derefence to get actual value
+    result.numMinBlocks = std::count(std::begin(blockSizes), std::end(blockSizes), result.minBlockSize);
+
+    result.maxBlockSize = *(minMaxSize.second); //Note have to derefence to get actual value
+    result.numMaxBlocks = std::count(std::begin(blockSizes), std::end(blockSizes), result.maxBlockSize);
+
+    
+    result.maxImbalance = static_cast<BoutReal>(result.maxBlockSize)/static_cast<BoutReal>(result.minBlockSize);
+
+    // Count the number of small blocks, defined as blocks less than smallSizeFrac of maxBlockSize
+    const BoutReal smallSizeFrac = 0.5;
+    result.numSmallBlocks = std::count_if(std::begin(blockSizes), std::end(blockSizes), [&blockSizes, &result, smallSizeFrac](int theSize) { return theSize < smallSizeFrac * result.maxBlockSize;}) ;
```

Comment:
Please could you clang-format this long line?

Reply:
I ran `git-clang-format-4.0` on this (after adding a trailing space so I had something to stage) and it said there was nothing to modify.

Reply:
I've manually split it as shown. Not sure why it wasn't being picked up automatically.